web/src/hooks/redux.js

```diff
@@ -102,3 +107,48 @@ export function useSmallLoaderVisible() {
   const windLoading = useSelector(state => state.data.isLoadingWind);
   return gridLoading || solarLoading || windLoading;
 }
+
+export function useCurrentNightTimes() {
+  const { zoneId } = useParams();
+  const zone = useSelector(state => state.data.grid.zones[zoneId]);
+  const datetimeStr = useSelector(state => state.data.grid.datetime);
+  const history = useCurrentZoneHistory();
+
+  return useMemo(() => {
+    if (!zone || !datetimeStr || !history || !history[0]) {
+      return [];
+    }
+    const { latitude, longitude } = getCenteredZoneViewport(zone);
+    const nightTimes = [];
+    let baseDatetime = moment(datetimeStr).startOf('day').toDate();
+    const earliest = history && history[0] && new Date(history[0].stateDatetime);
+    const latest = new Date((history && history[history.length - 1])
+      ? history[history.length - 1].stateDatetime
+      : datetimeStr);
+    do {
+      console.log('got it')
+      // Get last nightTime
+      const nightStart = getSunset(latitude, longitude, baseDatetime);
```

Comment:
No need to handle timezones ?

Reply:
Good question! 🤔 
We probably should take it into consideration, I'll look into it now

Reply:
Turns out, no - I spent a lot of time trying to adjust it, only to realise that the timeslider is in local time and since the sunset-sunrise library returns dates in UTC it shouldn't actually match the slider 😓 

Think this can be verified by changing to different timezones locally and checking that the sunrise/sunset times match reality:
![Screenshot 2021-07-21 at 02 12 29](https://user-images.githubusercontent.com/3296643/126455145-52cafeba-70d8-47d5-abe6-f1781266836f.png)

![Screenshot 2021-07-21 at 12 13 14](https://user-images.githubusercontent.com/3296643/126455137-233ad510-59e8-4c70-bc12-f447b005ea7b.png)